src/Nest/Mapping/Types/Core/Text/TextProperty.cs

```diff
@@ -8,7 +8,7 @@
 namespace Nest
 {
 	[JsonObject(MemberSerialization.OptIn)]
-	public interface ITextProperty : IProperty
+	public interface ITextProperty : ICoreProperty
```

Comment:
Will split DocValues out into a separate interface, and re-evaluate others as part of #2059